src/gui/tray/ActivityItem.qml

```diff
@@ -1,264 +1,99 @@
-import QtQml 2.12
-import QtQuick 2.9
-import QtQuick.Controls 2.2
+import QtQml 2.15
+import QtQuick 2.15
+import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 
 MouseArea {
-    id: activityMouseArea
+    id: root
 
     readonly property int maxActionButtons: 2
     property Flickable flickable
 
+    property bool isFileActivityList: false
+
+    property bool isChatActivity: model.objectType === "chat" || model.objectType === "room"
+
+    function activityLinksFiltered(activityLinks) {
+        // filter out the "dismiss" action for non-chat activities as per design (we will use a seperate X button for it)
+        if (root.isChatActivity) {
+            return activityLinks;
+        }
+
+        var filtered = activityLinks.filter(function(link) {
+            return link.verb === "DELETE";
+        });
+
+        return filtered;
+    }
+
+    property var linksForActionButtons: root.activityLinksFiltered(model.links)
```

Comment:
IMO, this could be more concise:

```
property var linksForActionButtons: root.isChatActivity ? model.links : model.links.filter(() => link.verb === "DELETE" )
```

Reply:
This should also eliminate some fo the funkiness inherent to setting bindings through functions